src/main/java/com/obj/nc/functions/processors/genericDataPersister/GenericDataPersister.java

```diff
@@ -0,0 +1,89 @@
+/*
+ * Copyright (C) 2021 the original author or authors.
+ * This file is part of Notiflow
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Lesser General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ * GNU Lesser General Public License for more details.
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this program. If not, see https://www.gnu.org/licenses/.
+ */
+
+package com.obj.nc.functions.processors.genericDataPersister;
+
+import com.fasterxml.jackson.databind.JsonNode;
+import com.obj.nc.domain.genericData.GenericData;
+import com.obj.nc.functions.processors.ProcessorFunctionAdapter;
+import com.obj.nc.repositories.GenericDataRepository;
+import lombok.RequiredArgsConstructor;
+
+import java.security.MessageDigest;
+import java.security.NoSuchAlgorithmException;
+import java.util.*;
+import java.util.stream.Collectors;
+
+@RequiredArgsConstructor
+public class GenericDataPersister extends ProcessorFunctionAdapter<List<JsonNode>, List<JsonNode>> {
+    private final GenericDataRepository genericDataRepository;
+    private final String externalId;
+
+    @Override
+    protected List<JsonNode> execute(List<JsonNode> newData) {
+        List<String> collectedIds = newData.stream().map(this::findIdByKey).collect(Collectors.toList());
```

Comment:
It would be much better for performance to have findIDsByKeys (not make N Sql call but just one)

Reply:
no, my bad - this is not call DB.. this is extracting ExternalIf from payload - thus a good name might be extractExtrnalIdFromPaylod

Reply:
renamed